### PR TITLE
show payment type if no payment method

### PR DESCRIPTION
**Pull request by adammeyer**
https://jira.cru.org/browse/EP-1855

### CODE REVIEW

**Comment by wrandall22** [comment]
So, I don't think this is quite what we want.  See comments on [EP-1827](https://jira.cru.org/browse/EP-1827). Summary: if `payment-method-link` is not there, then show `payment-type`.

**Comment by adammeyer** [comment]
@wrandall22 That still leaves some cells blank, is that okay?

**Comment by wrandall22** [comment]
Not really, no. There needs to be a task (or rolled into this one) that uses payment-method-link to get the inactive payment methods.

**Comment by adammeyer** [comment]
What was wrong with what I had?

**Comment by wrandall22** [comment]
What you currently have (if I'm reading it right) will show only the payment type for inactive payment methods instead of the `Card Type - Last Four` for CC and `Bank Name - Last Four` for EFT.

**Comment by adammeyer** [comment]
@OzzieOrca @Omicron7 could you provide your feedback on my latest commit before I write the specs?  Thanks

File: src/app/profile/yourGiving/recipientView/recipientGift/recipientGift.component.js [hunk 1]
```diff
@@ -31,22 +31,38 @@ class RecipientGift {
...
+        if(uri && paymentMethodUris.indexOf(uri) < 0){
+          paymentMethodUris.push(uri);
+          paymentMethodRequests.push(this.profileService.getPaymentMethod( uri, true ));
+        }
+      });
+
+      if(paymentMethodRequests.length){
+        Observable.forkJoin(paymentMethodRequests)
+          .subscribe((response) => {
+            angular.forEach(response, (paymentMethod) => {
```
Comment (OzzieOrca) [inline src/app/profile/yourGiving/recipientView/recipientGift/recipientGift.component.js]: You could avoid a nested loop if you use [keyBy](https://lodash.com/docs/4.17.4#keyBy) on the payment methods first and then for each donation, just look that value up in the object keyBy created.

**Comment by adammeyer** [comment]
@OzzieOrca Ready for your review again.

**Pull request merged**